include/prepared_statement.h

```diff
@@ -0,0 +1,42 @@
+#include <inttypes.h>
```

Comment:
Not sure where to best add this but to keep it in a comment I choose this.

I just tried this with the rust sqlx lib and it seems like there is something on one of the sides not quite going right. Queries fail with `encountered unexpected or invalid data: expecting ParseComplete but received ParameterDescription`. Not sure if this is a bug in here or in sqlx. But I thought it might be sensible to mention this in here as sqlx works fine with postgres itself just fine.

Reply:
Could you share a simple code sample that reproduces this? That will make it easier to debug

Reply:
So the query is already at https://github.com/launchbadge/sqlx/issues/67#issuecomment-1711977718 I will send the table I am creating when I am back at the pc 👍 (it's a little weird cross post since I was unsure where this came from. Might as well be sqlx being overly specific with responses it expects)

Reply:
The table in question looks like this:

```
CREATE TABLE IF NOT EXISTS users (
    username TEXT NOT NULL PRIMARY KEY UNIQUE,
    hash TEXT
);
```

Sadly I am not too deep into sqlx to know what they are doing exactly but from what I read they are using prepared statements internally to cache transactions. So there might be some more glue around the query itself thats happening here.

Reply:
Can you check again with the commit I pushed just now? I think it should be fixed.

Reply:
Just noticed also `WARNING S-0x1e434240: erooster/erooster@10.100.0.1:5432 got packet 'A' from server when not linked` in the pgbouncer logs. But it seems to not cause issues. So its more an observation.

Reply:
That sounds like you ran a LISTEN command. Those are not supported in PgBouncer in transaction mode (not even with this PR).

Reply:
Also there is an occasional `unexpected message: NotificationResponse` which I think is also related to this since I never saw it with normal postgres.

Reply:
Digging further that one seems to be caused by https://github.com/Diggsey/sqlxmq which uses Functions and sqlx. Especially I am seeing this for `mq_insert` which seems to performs an `PERFORM pg_notify(CONCAT('mq_', channel_name), '')` action to trigger other parts of its flow.

Reply:
> That sounds like you ran a LISTEN command. Those are not supported in PgBouncer in transaction mode (not even with this PR).

Whoops somehow github only after refresh showed me this comment. I will take a look if I can find it then :) Thanks for the hint

Reply:
yep its a LISTEN command apparently. Thanks for the hint and I will fix that on my code side then. The thread can be marked as resolved then.

Reply:
@MTRNord did you run the full test sqlx suite against PgBouncer or only some of your own code?